cpp/src/join/mixed_join_kernel.cuh

```diff
@@ -36,8 +36,10 @@ namespace detail {
 
 namespace cg = cooperative_groups;
 
+#pragma GCC diagnostic ignored "-Wattributes"
+
 template <cudf::size_type block_size, bool has_nulls>
-__launch_bounds__(block_size) __global__
+__attribute__((visibility("hidden"))) __launch_bounds__(block_size) __global__
```

Comment:
Why is this one different? (Not using `CUDF_KERNEL`) Same question for a few other mixed join kernels.

Reply:
My first pass at this was a simple regex and it marked up that specific template instantiations as well as the implementation. That failed, and I moved to  `__attribute__((visibility("hidden")))`.

Looking at it again, it seems like `CUDF_KERNEL` is entirely viable and I will switch to it

Reply:
Will need to go back to `visibility("hidden")` on the mixed join kernels. They are called across TUs and therefore can't be static:

```
File "/pyenv/versions/3.9.18/lib/python3.9/site-packages/cudf/_lib/__init__.py", line 4, in <module>
    from . import (
ImportError: /pyenv/versions/3.9.18/lib/python3.9/site-packages/cudf/_lib/../libcudf.so: undefined symbol: _ZN4cudf6detail15mixed_join_semiILi128ELb0EEEvNS_17table_device_viewES2_S2_S2_NS_12experimental3row4hash17device_row_hasherINS_7hashing6detail18MurmurHash3_x86_32ENS_7nullate7DYNAMICEEENS4_8equality31strong_index_comparator_adapterINSD_21device_row_comparatorILb0ESB_NSD_38nan_equal_physical_equality_comparatorEEEEENS0_9join_kindEN4cuco10static_mapIjiLN4cuda3std3__48__detail12thread_scopeE1EN3rmm2mr24stream_allocator_adaptorI17default_allocatorIcEEEE11device_viewEPiNS_3ast6detail22expression_device_viewEPKib
```

Reply:
> They are called across TUs and therefore can't be static:

This is highly suspicious. You shouldn't be able to call kernels across TU's without rdc enabled.

Reply:
> RDC controls device side function call visibility.

It also controls whether you can do:
```
extern __global__ void externally_defined_kernel();
```

Which is what I presumed you meant about calling kernels across TUs.

Did you just mean including the same `header.cuh` that defines a `__global__ void kernel()` and calling that kernel in multiple TUs?

Reply:
Turns out I was wrong. I was conflating this with the fact that you can't call kernels across DSO boundaries